src/sentry/models/projectteam.py

```diff
@@ -21,9 +21,8 @@ def get_for_teams_with_org_cache(self, teams: Sequence["Team"]) -> QuerySet["Pro
             .order_by("project__name", "project__slug")
             .select_related("project")
         )
-
-        # TODO(dcramer): we should query in bulk for ones we're missing here
-        orgs = {i.organization_id: i.organization for i in teams}
+        prefetch_related_objects(project_teams, "project__organization")
+        orgs = {pt.project.organization_id: pt.project.organization for pt in project_teams}
```

Comment:
Hmm, actually I was thinking you'd do this on the `teams`. But doing it on the `project_teams` actually does the caching you want in the next loop. So actually, you can just delete everything after line 24, since all that `set_cached_field_value` does is populate `project_team.project.organization`.

Reading this even more... We could also just do `.select_related("project", "project__organization")` and return that list of project teams. No idea why it wasn't done this way when originally implemented...

Reply:
yeah this makes sense. simplified.